File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day23OpeningTheTuringLockTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import advent.common.Pair;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day23OpeningTheTuringLockTest {
    @Test
    public void checkReal() {
        DailyProblem<Pair<Integer, Integer>, Pair<Integer, Integer>> dp = new Day23OpeningTheTuringLock("/twenty_fifteen/Day23-OpeningTheTuringLock-input.txt");

        assertEquals(184, dp.getPart1Answer().getSecond().intValue());
        assertEquals(231, dp.getPart2Answer().getSecond().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Pair<Integer, Integer>, Pair<Integer, Integer>> dp = new Day23OpeningTheTuringLock("/twenty_fifteen/Day23-OpeningTheTuringLock-example#1.txt");

        assertEquals(2, dp.getPart1Answer().getFirst().intValue());
        assertEquals(7, dp.getPart2Answer().getFirst().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day23OpeningTheTuringLockTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day23OpeningTheTuringLockTest  extends FlatSpec with Matchers {
  "2015 Day 23 - Input" should "calculate answers" in {
    val dp = new Day23OpeningTheTuringLock("/twenty_fifteen/Day23-OpeningTheTuringLock-input.txt")
    dp.part1Answer("b") should be(184)
    dp.part2Answer("b") should be(231)
  }

  "2015 Day 23 - Example #1" should "calculate answers" in {
    val dp = new Day23OpeningTheTuringLock("/twenty_fifteen/Day23-OpeningTheTuringLock-example#1.txt")
    dp.part1Answer("a") should be(2)
    dp.part2Answer("a") should be(7)
  }
}
